### PR TITLE
Exact table data

### PR DESCRIPTION
**Pull request by solomontang**
# Changes
The following props have been added to `Table`
- `exact :boolean` & `totalRows :number`
  - Use these properties if you only have the exact data to show per page, but still want to suggest the ability to paginate. Use the onPageChange/onSort callbacks to determine when to fetch fresh data.
- Add onPageChange
  - `
onPageChange ?:({
  column ?:string,
  order ?:'asc' | 'desc',
  page :number,
  rowsPerPage :number,
  start :number,
}) => void;
`
  - passed to `PaginationToolbar`

- Add onSort
  - `
onSort ?:({
  column ?:string,
  order ?:'asc' | 'desc',
  page :number,
  rowsPerPage :number,
  start :number,
}) => void;
`

image

### CODE REVIEW

File: src/table/src/components/TableBody.js [hunk 1]
```diff
@@ -26,14 +27,17 @@ const TableBody = (props :Props) => {
...
+    exact,
     order,
     orderBy,
     page,
     rowsPerPage,
   } = props;
 
   const sortedData = getSortedData(headers, data, order, orderBy);
-  const dataByPage :RowData[] = sortedData.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage);
+  const dataByPage :RowData[] = exact
```
Comment (giuliacampana) [inline src/table/src/components/TableBody.js]: So does the `exact` prop refer to a dataset that is already paginated?

File: src/table/stories/index.stories.js [hunk 2]
```diff
@@ -101,15 +104,96 @@ storiesOf('Table', module)
...
+
+    return (
+      <Card>
+        <CardSegment vertical>
+          <h1>exact + totalRows</h1>
+          <p>
+            {`
+              Use these properties if you only have the exact data to show per page, but still
+              want to suggest the ability to paginate. Use the onPageChange/onSort callbacks to
+              determine when to fetch fresh data.
```
Comment (giuliacampana) [inline src/table/stories/index.stories.js]: Oh I see—so by `exact` data, you mean that on each page data, you're fetching fresh data?

Comment (solomontang) [reply src/table/stories/index.stories.js]: That's the intention. You don't always want to fetch everything that could be described in a table at first render. If you're trying to show a table that has every issue, you shouldn't have to actually fetch all those issues - just the ones you're currently looking at. It used to be that the table would paginated based on all the data you have passed in. Now you can make the table show that there are 1000 rows while maintaining the same level of pagination, while only having the necessary data to show one page at a time.

exposing some callbacks with some basic overrides seems less misleading than making an "async" version of the table.

File: src/table/src/components/TableBody.js (hunk 1 above)
Comment (solomontang) [reply src/table/src/components/TableBody.js]: Yes.

**Pull request merged**